apps/openmw/mwclass/weapon.cpp

```diff
@@ -64,8 +66,9 @@ namespace MWClass
     bool Weapon::hasItemHealth (const MWWorld::ConstPtr& ptr) const
     {
         const MWWorld::LiveCellRef<ESM::Weapon> *ref = ptr.get<ESM::Weapon>();
+        ESM::WeaponType::Class weapClass = MWMechanics::getWeaponType(ref->mBase->mData.mType)->mWeaponClass;
 
-        return (ref->mBase->mData.mType < ESM::Weapon::MarksmanThrown); // thrown weapons and arrows/bolts don't have health, only quantity
+        return (weapClass == ESM::WeaponType::Melee || weapClass == ESM::WeaponType::Ranged); // thrown weapons and arrows/bolts don't have health, only quantity
```

Comment:
Maybe it's better to assume that anything that isn't ammunition/thrown weapon has condition. I'm not sure.

Reply:
Or maybe add a "hasHealth" flag to weapon type?